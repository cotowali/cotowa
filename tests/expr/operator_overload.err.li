type Word = string

fn (w1: Word) + (w2: Word, w3: Word): Word
fn (w1: Word) int |> + (w2: Word): Word
fn (w1: Word) ++ (w2: Word, w3: Word): Word
