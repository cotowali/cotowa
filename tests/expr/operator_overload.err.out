error: operator_overload.err.li 3,15: infix operator function must have 2 parameters
    3 | fn (w1: Word) == (): bool
      |               ^^
error: operator_overload.err.li 4,15: infix operator function must have 2 parameters
    4 | fn (w1: Word) + (w2: Word, w3: Word): Word
      |               ^
error: operator_overload.err.li 5,22: infix operator function cannot have pipe in
    5 | fn (w1: Word) int |> + (w2: Word): Word
      |                      ^
error: operator_overload.err.li 6,15: `++` is not infix operator
    6 | fn (w1: Word) ++ (w2: Word, w3: Word): Word
      |               ^^
